src/apps/aave/positions.ts

```diff
@@ -63,25 +64,63 @@ const hook: PositionsHook = {
       args: [aaveAddresses.poolAddressesProvider],
     })
 
-    const [userReserveData, userIncentivesData] = address
-      ? await client.multicall({
-          contracts: [
-            {
-              address: aaveAddresses.uiPoolDataProvider,
-              abi: uiPoolDataProviderV3Abi,
-              functionName: 'getUserReservesData',
-              args: [aaveAddresses.poolAddressesProvider, address as Address],
-            },
-            {
-              address: aaveAddresses.uiIncentiveDataProvider,
-              abi: uiIncentiveDataProviderV3Abi,
-              functionName: 'getUserReservesIncentivesData',
-              args: [aaveAddresses.poolAddressesProvider, address as Address],
-            },
-          ],
-          allowFailure: false,
-        })
-      : [undefined, undefined]
+    const [userReserveData, [reserveIncentiveData, _userIncentivesData]] =
+      address
+        ? await client.multicall({
+            contracts: [
+              {
+                address: aaveAddresses.uiPoolDataProvider,
+                abi: uiPoolDataProviderV3Abi,
+                functionName: 'getUserReservesData',
+                args: [aaveAddresses.poolAddressesProvider, address as Address],
+              },
+              {
+                address: aaveAddresses.uiIncentiveDataProvider,
+                abi: uiIncentiveDataProviderV3Abi,
+                functionName: 'getFullReservesIncentiveData',
+                args: [aaveAddresses.poolAddressesProvider, address as Address],
+              },
+            ],
+            allowFailure: false,
+          })
+        : [undefined, [undefined, undefined]]
+
+    // Note: Instead of calling `getAllUserRewards`, we could use reserveIncentiveData and userIncentivesData to get all user rewards
+    // but it requires some additional calculations to get the accrued rewards (linked to a/v/sToken held) on top of the unclaimedRewards to get total rewards.
+    // This is for simplicity here.
+    // See https://github.com/aave/aave-utilities/blob/446d9af6f14154771c0343538b59e2aeb7b38e47/packages/math-utils/src/formatters/incentive/calculate-all-user-incentives.ts#L31
+    const allUserRewards = await client.readContract({
+      address: aaveAddresses.incentivesController,
+      abi: incentivesControllerV3Abi,
+      functionName: 'getAllUserRewards',
+      args: [
+        // This builds the list of a/v/sToken address with incentives
+        // Because `getAllUserRewards` reverts if we pass an address with no incentives
+        reserveIncentiveData
```

Comment:
nit: looks like this code can be shared with the similar mapping done in shortcuts.

Reply:
ah yes, I was lazy, fixed 😄